frontend/public/components/monitoring/promql-expression-input.tsx

```diff
@@ -0,0 +1,395 @@
+        dom = dom.parentNode as HTMLElement
+      ) {
+        if (
+          dom.nodeName === 'LI' &&
+          (match = /-(\d+)$/.exec(dom.id)) &&
+          +match[1] < this.optionsLength
+        ) {
+          index = +match[1];
+          break;
+        }
+      }
+
+      if (index >= 0 && this.lastIndex !== index) {
+        this.lastIndex = index;
+        this.view.dispatch({ effects: setSelectedCompletion(index) });
+      }
+    }
+
+    destroy() {
+      this.view.dom.removeEventListener('mousemove', this.onMouseMove);
+    }
+  },
+);
+
+export const promqlHighlighter = HighlightStyle.define([
+  { tag: tags.name, color: '#000' },
+  { tag: tags.number, color: 'var(--pf-global--success-color--100)' },
+  { tag: tags.string, color: 'var(--pf-global--danger-color--200)' },
+  { tag: tags.keyword, color: 'var(--pf-global--default-color--200)', fontWeight: 'bold' },
+  {
+    tag: tags.function(tags.variableName),
+    color: 'var(--pf-global--default-color--200)',
+    fontWeight: 'bold',
+  },
+  { tag: tags.labelName, color: 'var(--pf-global--warning-color--200)' },
+  { tag: tags.operator },
+  { tag: tags.modifier, color: 'var(--pf-global--default-color--200)', fontWeight: 'bold' },
+  { tag: tags.paren },
+  { tag: tags.squareBracket },
+  { tag: tags.brace },
+  { tag: tags.invalid, color: 'red' },
+  { tag: tags.comment, color: '#888', fontStyle: 'italic' },
+]);
+
+export const PromQLExpressionInput: React.FC<PromQLExpressionInputProps> = ({
+  value,
+  onExecuteQuery,
+  onValueChange,
+}) => {
+  const { t } = useTranslation();
+  const containerRef = React.useRef<HTMLDivElement | null>(null);
+  const viewRef = React.useRef<EditorView | null>(null);
+  const [metricNames, setMetricNames] = React.useState<Array<string>>([]);
+  const [errorMessage, setErrorMessage] = React.useState<string | undefined>();
+
+  const placeholder = t('public~Expression (press Shift+Enter for newlines)');
+
+  const safeFetch = React.useCallback(useSafeFetch(), []);
+
+  React.useEffect(() => {
+    safeFetch(`${PROMETHEUS_BASE_PATH}/${PrometheusEndpoint.LABEL}/__name__/values`)
+      .then((response) => {
+        const metrics = response?.data;
+        setMetricNames(metrics);
+      })
+      .catch((err) => {
+        if (err.name !== 'AbortError') {
+          const message =
+            err?.response?.status === 403
+              ? t('public~Access restricted.')
+              : t('public~Failed to load metrics list.');
+          setErrorMessage(message);
+        }
+      });
+  }, [safeFetch, t]);
+
+  const onClear = () => {
+    if (viewRef.current !== null) {
+      const length = viewRef.current.state.doc.toString().length;
+      viewRef.current.dispatch({ changes: { from: 0, to: length } });
+    }
+    onValueChange('');
+  };
+
+  const onChange = (expressionValue: string) => {
+    if (expressionValue !== value) {
+      onValueChange(expressionValue);
+    }
+  };
+
+  const onExecute = React.useCallback(() => {
```

Comment:
It looks like `onExecute` is just wrapping `onExecuteQuery` here, so it looks to me like this doesn't add anything. In other words, can we just use `onExecuteQuery` directly in the `run` function below or am I missing something?